### PR TITLE
Migrate more `os.path` to `pathlib`

### PR DESCRIPTION
**Pull request by NotWearingPants**
## Overview: What does this pull request change?
Migrated more usages of `os.path` APIs to `pathlib`

## Motivation and Explanation: Why and how do your changes improve the library?
Further progress for #485

## Reviewer Checklist
- [ ] The PR title is descriptive enough for the changelog, and the PR is labeled correctly
- [ ] If applicable: newly added non-private functions and classes have a docstring including a short summary and a PARAMETERS section
- [ ] If applicable: newly added functions and classes are tested

### CODE REVIEW

File: manim/_config/utils.py [hunk 1]
```diff
@@ -21,7 +21,7 @@
 import sys
 from collections.abc import Mapping, MutableMapping
 from pathlib import Path
-from typing import Any, Iterable, Iterator
+from typing import Any, Iterable, Iterator, Optional
```
Comment (behackl) [inline manim/_config/utils.py:24]: Unused import?

File: manim/_config/utils.py [hunk 2]
```diff
@@ -85,7 +87,7 @@ def make_config_parser(custom_file: str = None) -> configparser.ConfigParser:
 
     Parameters
     ----------
-    custom_file : :class:`str`
+    custom_file : Union[:class:`str`, :class:`os.PathLike`], optional
```
Comment (behackl) [inline manim/_config/utils.py:90]: If there are actual type hints in the code, the type hints added in the docstring should be removed. (There are many places in the library where this has not been done yet.)

File: manim/mobject/svg/svg_mobject.py [hunk 3]
```diff
@@ -182,8 +182,7 @@ def generate_mobject(self) -> None:
         element_tree = ET.parse(file_path)
         new_tree = self.modify_xml_tree(element_tree)
         # Create a temporary svg file to dump modified svg to be parsed
-        root, ext = os.path.splitext(file_path)
-        modified_file_path = root + "_" + ext
+        modified_file_path = modified_file_path.with_stem(file_path.stem + "_")
```
Comment (behackl) [inline manim/mobject/svg/svg_mobject.py:185]: This currently causes tests to fail, you probably want something like 
```suggestion
        modified_file_path = file_path.with_stem(file_path.stem + "_")
```

**Review by behackl — COMMENTED** [review]
I've coarsely read over the changes, they look good to me in general; thank you! Seems like you've fixed the problem with the SVG file path already too. Thanks for looking into this!

File: manim/mobject/svg/svg_mobject.py (hunk 3 above)
Comment (NotWearingPants) [reply manim/mobject/svg/svg_mobject.py]: Apparently `with_stem` is new from 3.9, and the read-the-docs build runs on 3.8

Comment (behackl) [reply manim/mobject/svg/svg_mobject.py]: True, then it will have to be rewritten.

**Comment by NotWearingPants** [comment]
Thanks for the review! There are more usages of `os.path` (specifically in tests) that I plan to convert, do you prefer I update this PR or open separate ones?

**Comment by NotWearingPants** [comment]
I made the changes to `tests` in a separate PR (#2991), hopefully makes it easier to review

File: manim/utils/tex_file_writing.py [hunk 4]
```diff
@@ -211,22 +210,19 @@ def convert_to_svg(dvi_file, extension, page=1):
...
-    result = dvi_file.replace(extension, ".svg")
-    result = Path(result)
-    result_str = result.as_posix()
-    dvi_file = Path(dvi_file).as_posix()
+    result = dvi_file.with_suffix(".svg")
     if not result.exists():
         commands = [
             "dvisvgm",
-            "--pdf" if extension == ".pdf" else "",
+            "--pdf" if dvi_file.suffix == ".pdf" else "",
```
Comment (behackl) [inline manim/utils/tex_file_writing.py:220]: I think this change should be made separately (along with refactoring the variable name from `dvi_file` to something that captures that the file can be an xdv or pdf as well).

```suggestion
            "--pdf" if extension == ".pdf" else "",
```

**Review by behackl — CHANGES_REQUESTED** [review]
Thank you very much for your efforts! This looks good to me in general, please let me know whether you can take care of (or want to discuss) my requested changes -- otherwise I think this can be merged.

**Review by behackl — APPROVED** [review]

**Pull request merged**